### ISSUE TITLE
[Bug]: Cannot update listening port via qBittorrent JSON

### ISSUE THREAD

**Issue by pyrrhichios** [VueTorrent/VueTorrent#1984]
### Bug description

Cannot use the following script line to update the qBittorrent listening port when VueTorrent is the active WebUI:

curl -s -b /tmp/qbitcookie.txt "http://localhost:PORT/api/v2/app/setPreferences" -d 'json={"listen_port": "'"$portforward"'"}'

(PORT obscured)

$portforward environment variable is set correctly. When VueTorrent is deactivated, this command will work and update the listening port within the vanilla WebUI interface.

### Expectations

Standard JSON update form should work with VueTorrent as active WebUI.

### Relevant debug informations

No errors seen in console.

### Steps to reproduce

1. ...Set a a listening port environment variable ($portforward)
2. ...Send port to qBitorrent using curl -s -b /tmp/qbitcookie.txt "http://localhost:PORT/api/v2/app/setPreferences" -d 'json={"listen_port": "'"$portforward"'"}' replacing PORT with the WebUI port.
3. Check listening port in qBittorrent settings - port will not be updated with the pushed port.

### How did you install VueTorrent?

Downloaded from GitHub releases

### Environment description

```plain text
Device type: (Desktop / Laptop / Tablet / Phone)
OS: (Windows / Android / iOS / macOS / ...)
Browser (name and version): 

PWA: Yes / No
Running inside a container: Yes / No
```

### qBittorrent version

4.6.7

### VueTorrent version

2.12.0

### Agreements

- [X] The bug still occurs on the nightly build
- [X] The bug still occurs on the latest version of qBittorrent
- [X] A similar issue doesn't already exist
- [X] I checked the DevTools and included any errors that may show up

**Comment by pyrrhichios** [comment]
Sorry, I can't seem to edit this - VueTorrent updated to 2.15.0 and issue persists.

**Comment by Larsluph** [comment]
VueTorrent is only a WebUI so calling the qBittorrent API directly is not relevant here.

The `listen_port` value is an integer, make sure you remove any quotation marks in the json you send.

**Comment by pyrrhichios** [comment]
> VueTorrent is only a WebUI so calling the qBittorrent API directly is not relevant here.
> 
> The `listen_port` value is an integer, make sure you remove any quotation marks in the json you send.

Thanks for your reply. I've changed the command to this:

curl -s -b /tmp/qbitcookie.txt -H "Content-Type: application/json" "http://localhost:7056/api/v2/app/setPreferences" -d "{\"listen_port\":$portforward}"

(ChatGPT-fu I'm afraid)

Again, this works just fine when I drop to the regular WebUI interface, but not when Vuetorrent is active. I'm really rather baffled. Any ideas?

**Comment by Larsluph** [comment]
Generated from the code, here's the command we use converted to cURL:
```sh
curl -s -b /tmp/qbitcookie.txt -X POST -H 'content-type: application/x-www-form-urlencoded' -d 'json={"listen_port": $portforward}' 'http://localhost:PORT/api/v2/app/setPreferences'
```

**Comment by pyrrhichios** [comment]
> Generated from the code, here's the command we use converted to cURL:
> 
> ```shell
> curl -s -b /tmp/qbitcookie.txt -X POST -H 'content-type: application/x-www-form-urlencoded' -d 'json={"listen_port": $portforward}' 'http://localhost:PORT/api/v2/app/setPreferences'
> ```

Thank you so much for taking the time to do this, much appreciated!

So: I think I may have misdiagnosed the problem - I think this command and previous ones probably all worked just fine, but there's something odd going on with Vuetorrent not showing the current listening port without a hard refresh of the browser window. 

To explain, I tried substituting in your command to my script, and again, it worked as expected in standard WebUI mode, then I switched to Vuetorrent - no dice. Then it occurred to me to check whether the listening port had actually been updated in the qBittorrent.conf file, and sure enough it was set to the right one. This inspired me to try hard refreshing the chrome tab, i.e. hold ctrl and click refresh, and this time on opening the settings menu it was displaying the correct port.

I'm not sure if this is a genuine issue with how Vuetorrent is cached that other users might see or just indicating some other weirdness in my setup, but I thought I'd report back. 

Thanks again!

**Comment by Larsluph** [comment]
A lot of data provided by qBittorrent are stored in your sessionStorage, which is a dedicated storage for a single tab. At every launch (on startup and on refresh) we fetch the current configuration to update this cache.

Can you please check the `preferences` request from qBittorrent and see if the `listen_port` correctly matches to the one in the `qBittorrent.conf`?

**Comment by pyrrhichios** [comment]
> A lot of data provided by qBittorrent are stored in your sessionStorage for caching, which is a dedicated storage in your browser created for a single tab (called "session"). At every launch (on startup and on refresh) we fetch the current configuration to update this cache.
> 
> Can you please check the `preferences` request from qBittorrent and see if the `listen_port` correctly matches to the one in the `qBittorrent.conf`?

Thanks for getting back to me! - Happy to help with chasing this down if you think this is a problem that will affect other users. I'm afraid I'm not a power user when it comes to these things though - if you step me through where to look etc. I'd be glad to give it a shot?

**Comment by Larsluph** [comment]
We have a [wiki page](https://github.com/VueTorrent/VueTorrent/wiki/Support#how-to-view-network-requests) for this

**Comment by pyrrhichios** [comment]
> We have a [wiki page](https://github.com/VueTorrent/VueTorrent/wiki/Support#how-to-view-network-requests) for this

Thanks - what I did was change the listen port to something else, then run my script that updates the listen port. I had the developer tools open in Chrome, and I was on the "Network" tab of it with the "Fetch/XHR" filter on. Is that where I should be looking?

If so - I don't see the 'preferences' network event when I run my script that should update qBittorrent automatically - only when I use the interface and click the save button. The listen port in preference events initiated from the interface do indeed match and update normally.

**Comment by Larsluph** [comment]
Preferences on VueTorrent are only fetched after an update on on app startup, so to see the request you have to reload your browser tab after your script.

If the listen_port value you see in that request matches the one displayed on VueTorrent it may be a sign that your script doesn't work...

**Comment by pyrrhichios** [comment]
> Preferences on VueTorrent are only fetched after an update on on app startup, so to see the request you have to reload your browser tab after your script.
> 
> If the listen_port value you see in that request matches the one displayed on VueTorrent it may be a sign that your script doesn't work...

Thanks - if you have to reload your browser tab to see the update then I think what I'm seeing is expected behaviour then, i.e. the script is indeed updating the port, but Vuetorrent UI will continue to show the old port until the browser tab is hard refreshed.

I checked this out in the console and it is indeed as you say - when the port is updated by the script there's no updated preferences request to check until the tab is refreshed - at which point there is a preferences request call, which does indeed show the port the script sent.

In short - all well I think, as long as users appreciate that they may need to reload a tab to see up to date settings set by something other than the web UI, which is probably a pretty uncommon scenario...

**Comment by Larsluph** [comment]
As the server doesn't send anything we have to explicitly fetch them. To prevent excessive bandwidth usage, the app assumes that no external interactions are made.
